Update Transport transactions documentation (#6461)
Particular/docs.particular.net: transports/transactions.md

#### Atomicity and consistency guarantees

> In the Distributed Transaction mode, handlers will execute inside a `TransactionScope` created by the transport. This means that all the data updates and queue operations are committed or rolled back.

Reviewer: ```suggestion
When the `TransportTransactionMode` is set to `TransactionScope`, handlers will execute inside a `System.Transactions.TransactionScope` created by the transport. This means that all the data updates and queue operations are committed or rolled back.
```

NOTE: This mode requires the selected storage to support participating in distributed transactions.

### Transport transaction - Sends atomic with Receive

Some transports support enlisting outgoing operations in the current receive transaction. This prevents messages from being sent to downstream endpoints during retries.

Use the following code to use this mode:

snippet: TransportTransactionAtomicSendsWithReceive

> 
> 

Reviewer: ```suggestion

```
